core/src/main/java/de/bwaldvogel/mongo/backend/aggregation/Expression.java

```diff
@@ -163,7 +163,9 @@ Object apply(List<?> expressionValue, Document document) {
             if (index < 0 || index >= collection.size()) {
                 return null;
             } else {
-                return collection.get(index);
+                Object ret = collection.get(index);
+                if(Missing.isNullOrMissing(ret)) return null;
```

Comment:
Actually the handling of "missing" and null values was not quite correct.
I’ve added an integration test in d44ca92b208def9e74f103d7e59bf17e388fff18 which compares the behavior to a real MongoDB.
I’ve adjusted the implementation to behave exactly like MongoDB does.

Reply:
Oh right- must confess I didn't test this edge case vs mongo, I just assumed it had to work like the existing tests 😮 

[I wonder if one could construct a test framework that runs the very same tests vs an actual mongo to check it's behaving the same?]

Reply:
> I wonder if one could construct a test framework that runs the very same tests vs an actual mongo to check it's behaving the same?

That’s exactly what we already do :)

- `AbstractBackendTest` has multiple derived classes:
    1. `RealMongoBackendTest`
    2. `MemoryBackendTest`

Basically the same for `AbstractAggregationTest`
1. `RealMongoAggregationTest`
2. `MemoryBackendAggregationTest`